karton/core/config.py

```diff
@@ -34,34 +34,153 @@ class Config(object):
+        ...
+
+    def getint(
+        self, section_name: str, option_name: str, fallback: Optional[int] = None
+    ) -> Optional[int]:
+        """
+        Gets value from configuration or returns ``fallback`` (None by default)
+        if value was not set. Value is coerced to int type.
+        """
+        value = self.get(section_name, option_name, fallback)
+        if value is None:
+            return None
+        return int(value)
+
+    @overload
+    def getboolean(self, section_name: str, option_name: str, fallback: bool) -> bool:
+        ...
+
+    @overload
+    def getboolean(self, section_name: str, option_name: str) -> Optional[bool]:
+        ...
+
+    def getboolean(
+        self, section_name: str, option_name: str, fallback: Optional[bool] = None
+    ) -> Optional[bool]:
+        """
+        Gets value from configuration or returns ``fallback`` (None by default)
+        if value was not set. Value is coerced to bool type.
+
+        See also:
+        https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean
+        """
+        value = self.get(section_name, option_name, fallback)
+        if value is None:
+            return None
+        if type(value) is bool:
+            return value
+        if type(value) is str and value.lower() in ["1", "yes", "true", "on"]:
+            return True
+        elif type(value) is str and value.lower() in ["0", "no", "false", "off"]:
+            return False
+        else:
+            raise ValueError(f"{section_name}.{option_name} is not a correct boolean")
+
+    def append_to_list(self, section_name: str, option_name: str, value: Any):
+        """
+        Appends value to a list in configuration
+        """
+        if section_name not in self._config:
+            self._config[section_name] = {}
+        if option_name not in self._config[section_name]:
+            self._config[section_name][option_name] = []
+        elif not isinstance(self._config[section_name][option_name], list):
+            raise TypeError(
+                f"{section_name}.{option_name} is "
+                f"{type(self._config[section_name][option_name])} while "
+                f"list was expected"
+            )
+        self._config[section_name][option_name].append(value)
+
+    def load_from_dict(self, data: Dict[str, Dict[str, Any]]):
+        """
+        Updates configuration values from dictionary compatible with
+        ``ConfigParser.read_dict``. Accepts value in native type, so you
+        don't need to convert them to string.
+
+        None values are treated like missing value and are not added.
+
+        .. code-block::
+        {
+           "section-name": {
+               "option-name": "value"
+           }
+        }
+        """
+        for section_name, section in data.items():
+            for option_name, value in section.items():
+                if value is None:
```

Comment:
Not really because configparser throws exception in case of None values. 

I decided to treat None as missing key, because it works well with argparse: default value for missing arguments is None. And getter methods are returning None when the key is missing as well.